src/multilinear.jl

```diff
@@ -136,55 +204,31 @@ for (T, wrapper) ∈ [(:AbstractMatrix, :identity), (:(Hermitian), :(Hermitian))
             isempty(transp) && return X
             length(transp) == length(dims) && return $wrapper(collect(transpose(X)))
 
-            keep = Vector{eltype(transp)}(undef, length(dims) - length(transp)) # Systems kept
-            counter = 0
-            for i ∈ 1:length(dims)
-                if i ∉ transp
-                    counter += 1
-                    keep[counter] = i
-                end
-            end
-
-            d = size(X, 1)                            # Dimension of the final output Y
-            Y = similar(X, (d, d))                    # Final output Y
+            nsys = length(dims)
+            keep = _subsystems_complement(transp, nsys)
 
-            tXi = Vector{Int}(undef, length(dims))    # Tensor indexing of X for row
-            tXj = Vector{Int}(undef, length(dims))    # Tensor indexing of X for column
+            dims_keep = dims[keep]
+            dims_transp = dims[transp]
 
-            tYi = Vector{Int}(undef, length(dims))    # Tensor indexing of Y for row
-            tYj = Vector{Int}(undef, length(dims))    # Tensor indexing of Y for column
+            keep_size = prod(dims_keep)
+            transp_size = prod(dims_transp)
+            prod(dims_keep) > prod(dims_transp) && return partial_transpose(transpose(X), keep, dims)
 
-            @inbounds for j ∈ 1:d
-                _tidx!(tYj, j, dims)
-                for i ∈ 1:j-1
-                    _tidx!(tYi, i, dims)
+            X_size = size(X, 1)                            # Dimension of the final output Y
+            Y = similar(parent(X))                    # Final output Y
```

Comment:
```suggestion
            Y = similar(X, X_size, X_size)                    # Final output Y
```
as it turns out I introduced a bug here, sorry about that